PixelHunter1995/Components/SpeechLine.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+
+namespace PixelHunter1995.Components
+{
+    internal class SpeechLine : IUpdateable
+    {
+        private string ActiveSpeech;
+        private TimeSpan SpeechEndTime;
+        public bool Speaking { get; private set; }
+        private bool ShouldStartNextLine;
+        private readonly Queue UpcomingLines;
+        private static readonly int TALKING_SPEED = 100;  // Number of milliseconds per character in line
+        private static readonly int CHARS_PER_LINE = 40;
+
+        public SpeechLine()
+        {
+            ActiveSpeech = "";
+            Speaking = false;
+            ShouldStartNextLine = false;
+            UpcomingLines = new Queue();
+        }
+
+        public void Say(string speech)
+        {
+            ShouldStartNextLine = true;
+            UpcomingLines.Enqueue(speech);
+        }
+
+        public void Update(GameTime gameTime)
+        {
+            if (!Speaking || ShouldStartNextLine)
+            {
+                if (UpcomingLines.Count != 0)
+                {
+                    NextLine(gameTime);
+                }
+                else
+                {
+                    Speaking = false;
+                }
+            }
+            else if (SpeechEndTime < gameTime.TotalGameTime)  // Done with this line
+            {
+                ShouldStartNextLine = true;
+            }
+        }
```

Comment:
You don't need many `if` statements to confuse me... Would the code below do approximately the same thing?

```csharp
if (!Speaking)
{
    return;
}

if (UpcomingLines.Count == 0)
{
    Speaking = false;
    return;
}

bool shouldStartNextLine = SpeechEndTime < gameTime.TotalGameTime;
if (shouldStartNextLine)
{
    NextLine(gameTime);
}
```

Only difference I can directly see is that a next line will come the same frame that the `SpeechEndTime` is reached instead of the next frame.

Reply:
```
if (UpcomingLines.Count == 0)
{
    return;
}

bool shouldStartNextLine = SpeechEndTime < gameTime.TotalGameTime;
if (shouldStartNextLine)
{
    NextLine(gameTime);
}
```

Reply:
```csharp
public bool IsSpeaking()
{
    return UpcomingLines.Count > 0;
}
```